src/components/MainNav/MainNav.js

```diff
@@ -86,6 +99,39 @@ class MainNav extends Component {
     this.context.router.history.push('/');
   }
 
+  subscribeToQueryChanges(moduleInfo) {
+    if (this.unsub) this.unsub();
+    if (!moduleInfo.queryResource) return;
+    this.unsub = this.store.subscribe(() => {
+      const previousQueryValues = this.queryValues;
+      // This is not DRY, as it was expressed already in LocalResource is stripes-connect,
+      // And in Root.js in stripes-core. Both State Keys should be derived from a common mechanism.
+      this.queryValues = this.store.getState()[`${moduleInfo.dataKey ? `${moduleInfo.dataKey}#` : ''}${_.snakeCase(moduleInfo.module)}_${moduleInfo.queryResource}`];
```

Comment:
Only apps can surface resources in the query. Apps aren't connected with a dataKey.